packages/gatsby/src/utils/changed-pages.ts

```diff
@@ -0,0 +1,54 @@
+import { boundActionCreators } from "../redux/actions"
+const { deletePage, deleteComponentsDependencies } = boundActionCreators
+
+import { isEqualWith, IsEqualCustomizer } from "lodash"
+import { IGatsbyPage } from "../redux/types"
+
+export function deleteUntouchedPages(
+  currentPages: Map<string, IGatsbyPage>,
+  timestamp: number
+): string[] {
+  const deletedPages: string[] = []
+
+  // Delete pages that weren't updated when running createPages.
+  currentPages.forEach(page => {
+    if (
+      !page.isCreatedByStatefulCreatePages &&
+      page.updatedAt < timestamp &&
+      page.path !== `/404.html`
+    ) {
+      deleteComponentsDependencies([page.path])
+      deletePage(page)
+      deletedPages.push(page.path, `/page-data${page.path}`)
+    }
+  })
+  return deletedPages
+}
+
+export function findChangedPages(
+  oldPages: Map<string, IGatsbyPage>,
+  currentPages: Map<string, IGatsbyPage>
+): {
+  changedPages: string[]
+  deletedPages: string[]
+} {
+  const changedPages: string[] = []
+
+  const compareWithoutUpdated: IsEqualCustomizer = (_left, _right, key) =>
+    key === `updatedAt` || undefined
+
+  currentPages.forEach((newPage, key) => {
+    const oldPage = oldPages.get(key)
+    if (!oldPage || !isEqualWith(newPage, oldPage, compareWithoutUpdated)) {
```

Comment:
Why not do this instead?
```suggestion
    if (!oldPage || key === 'updatedAt' || _.isEqual(newPage, oldPage)) {
```

(I suspect that you could even drop the `!oldPage` condition as it looks like it's subsumed by the `isEqual` check.)

Reply:
That won't work. It's confused by the different meanings of `key`. On line 42 it's the page id, but on 37 it's the field in the object. I'll use better names

Reply:
In this case I would suggest renaming `key` here to have some semantic meaning - `pagePath` maybe?